connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -241,43 +240,26 @@ public ConfigInfos validateConnectorConfig(Map<String, String> connectorConfig)
         Connector connector = getConnector(connType);
         ClassLoader savedLoader = plugins().compareAndSwapLoaders(connector);
         try {
-            // do basic connector validation (name, connector type, etc.)
-            ConfigDef basicConfigDef = (connector instanceof SourceConnector)
-                                       ? SourceConnectorConfig.configDef()
-                                       : SinkConnectorConfig.configDef();
+            ConfigDef baseConfigDef = (connector instanceof SourceConnector)
+                    ? SourceConnectorConfig.configDef()
+                    : SinkConnectorConfig.configDef();
+            ConfigDef enrichedConfigDef = ConnectorConfig.enrich(plugins(), baseConfigDef, connectorProps, false);
             Map<String, ConfigValue> validatedConnectorConfig = validateBasicConnectorConfig(
                     connector,
-                    basicConfigDef,
-                    connectorConfig
+                    enrichedConfigDef,
+                    connectorProps
             );
             configValues.addAll(validatedConnectorConfig.values());
-            configKeys.putAll(basicConfigDef.configKeys());
-            allGroups.addAll(basicConfigDef.groups());
-
-            ConnectorConfig connectorConfigToEnrich = (connector instanceof SourceConnector)
-                    ? new SourceConnectorConfig(plugins(), connectorConfig)
-                    : new SinkConnectorConfig(plugins(), connectorConfig);
-            final ConfigDef connectorConfigDef = connectorConfigToEnrich.enrich(
-                    plugins(),
-                    basicConfigDef,
-                    connectorConfig,
-                    false
-            );
-
-            // Override is required here after the enriched ConfigDef has been created successfully
-            configKeys.putAll(connectorConfigDef.configKeys());
-            allGroups.addAll(connectorConfigDef.groups());
+            configKeys.putAll(enrichedConfigDef.configKeys());
+            allGroups.addAll(enrichedConfigDef.groups());
 
             // do custom connector-specific validation
-            Config config = connector.validate(connectorConfig);
+            Config config = connector.validate(connectorProps);
             ConfigDef configDef = connector.config();
             configKeys.putAll(configDef.configKeys());
             allGroups.addAll(configDef.groups());
             configValues.addAll(config.configValues());
             return generateResult(connType, configKeys, configValues, new ArrayList<>(allGroups));
-        } catch (ConfigException e) {
```

Comment:
Nothing should be throwing it now since we don't construct any subclasses of AbstractConfig. The validation code shouldn't be throwing ConfigException. I think this was an artifact of the previous iteration where constructing the Source/SinkConnectorConfig could potentially throw that exception because it was doing the normal parsing instead of "validation parsing".